packages/flutter/lib/src/rendering/viewport.dart

```diff
@@ -1245,7 +1245,7 @@ abstract class RenderViewportBase<ParentDataClass extends ContainerParentDataMix
       trailingEdgeOffset: trailingEdgeOffset,
       currentOffset: currentOffset,
     );
-    if (targetOffset == null) {
+    if (targetOffset == null || !targetOffset.offset.isFinite) {
```

Comment:
I think the viewport should not require a change here, since the issue lies in the sliver.

Reply:
Would the comment below here still be accurate with this change?

Reply:
Can you share an example of that? An infinitely long sliver that precedes a text field would mean the text field is absolutely inaccessible.

Reply:
```dart
import 'package:flutter/material.dart';

FocusNode focusNode = FocusNode();
void main() {
  runApp(const MyApp());
}

class MyApp extends StatelessWidget {
  const MyApp({super.key});

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      home: Scaffold(
        body: CustomScrollView(
          slivers: [
            SliverMainAxisGroup(
              slivers: [
                SliverList.builder(
                  itemBuilder: (context, index) {
                    return Container(
                      height: 44,
                    );
                  },
                ),
                SliverToBoxAdapter(
                  child: TextField(
                    focusNode: focusNode,
                  ),
                ),
              ],
            ),
          ],
        ),
        floatingActionButton: IconButton(
          onPressed: focusNode.requestFocus,
          icon: const Icon(Icons.add),
        ),
      ),
    );
  }
}
```
Like this.

Reply:
Thanks! Yes, this would be a case where I would expect the SliverMainAxisGroup to assert because the infinite sliver does not allow the following sliver in the group to be seen. It is an issue in the group rather than the viewport.

Reply:
Do you mean to assert in the `childScrollOffset` method of `SliverMainAxisGroup`? `childScrollOffset` returning Infinite should be the correct behavior. As I mentioned before, this is not just a problem with `SliverMainAxisGroup`. In the code above, crashes will occur even if we do not use `SliverMainAxisGroup` and put the sliver directly in the `CustomScrollView`. I will remove the assertion, it introduces some complex issues that may not be within the scope of this PR.

Reply:
I have submitted a new issue for it. #143047.

Reply:
> Do you mean to assert in the childScrollOffset method of SliverMainAxisGroup?

No, probably in RenderSliverMainAxisGroup.performLayout. It is not valid for a child of the group to have an infinite extent if there is another child following behind it. You would never be able to scroll to that following child.

Reply:
Thanks! If the assertion is in `RenderSliverMainAxisGroup.performLayout`, then does `RenderViewportBase.layoutChildSequence` also need an assertion? It might not be appropriate to do this in this PR. Also, since English is not my native language, I might not be very good at providing debug messages. Could you please do this?

Reply:
> Could you please do this?

Do you mean you would like to abandon this PR?

I can help with the error message from here on Github if you would like to add the assertion to RenderSliverMainAxisGroup.performLayout. :)

I don't think we should add the assertion to the viewport, since this issue is specific to the SliverMainAxisGroup. That is the issue we are trying to fix. If we have a case that supports adding it in the viewport, we should file a separate bug and PR so we can understand what that scenario is.

Reply:
I'm very sorry, my translation may have caused some misunderstandings. I've added an assertion, please continue with the review.

Reply:
No apologies needed. Thank you for continuing to work on this! :)